lib/logstash/inputs/http.rb

```diff
@@ -75,11 +83,20 @@ def register
   end # def register
 
   def run(queue)
```

Comment:
You can provide a block to the Auth::Basic to support multiple credentials.
```ruby
use Rack::Auth::Basic, "secret" do |username, password|
  username == 'foo' && password == 'bar'
end
```

I don't think its overkill, it gives us flexibility and rack we will get security upgrades when they come.